In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import r2_score
from sklearn.linear_model import LogisticRegression

# LINEAR REGRESSION

> Indented block



In [5]:
df= pd.read_csv('/content/car_age_price.csv')
df.head()

Year   Price
0  2018  465000
1  2019  755000
2  2019  700000
3  2018  465000
4  2018  465000

In [6]:
print('No of rows:',df.shape[0])
print('No of columns:',df.shape[1])

No of rows: 112
No of columns: 2


In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112 entries, 0 to 111
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   Year    112 non-null    int64
 1   Price   112 non-null    int64
dtypes: int64(2)
memory usage: 1.9 KB


In [8]:
X,y= df['Year'],df['Price']

In [9]:
X_train,X_test, y_train, y_test= train_test_split(X,y,test_size= 0.2,random_state= 10)

In [10]:
model= LinearRegression()

In [11]:
model.fit(X_train[:, np.newaxis],y_train)

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Support for multi-dimensional indexing (e.g. `obj[:, None]`) is deprecated and will be removed in a future version.  Convert to a numpy array before indexing instead.
  """Entry point for launching an IPython kernel.


LinearRegression()

In [12]:
X_test_reshaped= X_test.values.reshape(-1,1)
pred= model.predict(X_test_reshaped)

In [13]:
print(pred)

[500555.87807959 500555.87807959 500555.87807959 455459.31838281
 320169.63929248 320169.63929248 545652.43777637 545652.43777637
 590748.99747315 635845.55716993 545652.43777637 410362.75868604
 500555.87807959 365266.19898926 590748.99747315 590748.99747315
 500555.87807959 545652.43777637 410362.75868604 590748.99747315
 500555.87807959 500555.87807959 500555.87807959]


In [14]:
print("r2 score:", r2_score(pred, y_test))

r2 score: 0.34112487409755354


In [15]:
print(model.predict([[2022]]))

[726038.67656349]


**Here we can see that for the year 2022 it tpredicts 726038.67656349**

# LOGISTIC REGRESSION

In [16]:
dt= pd.read_csv('/content/Iris.csv')
dt.head()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  Species
0   1            5.1           3.5            1.4           0.2        0
1   2            4.9           3.0            1.4           0.2        0
2   3            4.7           3.2            1.3           0.2        0
3   4            4.6           3.1            1.5           0.2        0
4   5            5.0           3.6            1.4           0.2        0

In [17]:
print('No of rows:',dt.shape[0])
print('No of columns:',dt.shape[1])

No of rows: 100
No of columns: 6


In [18]:
dt.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             100 non-null    int64  
 1   SepalLengthCm  100 non-null    float64
 2   SepalWidthCm   100 non-null    float64
 3   PetalLengthCm  100 non-null    float64
 4   PetalWidthCm   100 non-null    float64
 5   Species        100 non-null    int64  
dtypes: float64(4), int64(2)
memory usage: 4.8 KB


In [19]:
dt.describe()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
count  100.000000     100.000000    100.000000     100.000000    100.000000   
mean    50.500000       5.471000      3.094000       2.862000      0.785000   
std     29.011492       0.641698      0.476057       1.448565      0.566288   
min      1.000000       4.300000      2.000000       1.000000      0.100000   
25%     25.750000       5.000000      2.800000       1.500000      0.200000   
50%     50.500000       5.400000      3.050000       2.450000      0.800000   
75%     75.250000       5.900000      3.400000       4.325000      1.300000   
max    100.000000       7.000000      4.400000       5.100000      1.800000   

          Species  
count  100.000000  
mean     0.500000  
std      0.502519  
min      0.000000  
25%      0.000000  
50%      0.500000  
75%      1.000000  
max      1.000000

In [20]:
X,y= dt['SepalLengthCm'],dt['Species']

In [21]:
X_train,X_test, y_train, y_test= train_test_split(X,y,test_size= 0.2,random_state= 10)

In [22]:
Lr_model= LogisticRegression()

In [23]:
Lr_model.fit(X_train[:, np.newaxis],y_train)

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Support for multi-dimensional indexing (e.g. `obj[:, None]`) is deprecated and will be removed in a future version.  Convert to a numpy array before indexing instead.
  """Entry point for launching an IPython kernel.


LogisticRegression()

In [24]:
X_test_reshaped= X_test.values.reshape(-1,1)

In [25]:
pred= model.predict(X_test_reshaped)

In [26]:
print(pred)

[-90229212.57586858 -90197644.98408084 -90233722.23183826
 -90238231.88780794 -90206664.2960202  -90251760.85571697
 -90202154.64005052 -90256270.51168665 -90260780.16765633
 -90179606.36020212 -90247251.1997473  -90238231.88780794
 -90233722.23183826 -90211173.95198987 -90202154.64005052
 -90170587.04826277 -90197644.98408084 -90233722.23183826
 -90224702.9198989  -90229212.57586858]


In [27]:
print(Lr_model.predict([[7.5]]))

[1]


from this we can infer that sepal length of 7.5cm is belong to 1, ie iris versicolor


